pandas/core/internals/base.py

```diff
@@ -49,6 +52,11 @@
     )
 
 
+class _AlreadyWarned:
+    def __init__(self):
+        self.warned_already = False
```

Comment:
Can you add a small comment here about why such a class is needed (in contrast to a simple boolean argument)?

Reply:
added